HEN_HOUSE/egs++/ausgab_objects/egs_fluence_scoring/egs_fluence_scoring.cpp

```diff
@@ -0,0 +1,1637 @@
+                 Emax = flu_s ? exp(flu_xmax):flu_xmax;
+       sprintf(buf,"%g MeV and %g MeV energy range",Emin,Emax);
+       description += buf;
+       if (flu_s)
+          description += " on a logarithmic scale \n";
+       else
+          description += " on a linear scale \n";
+    }
+}
+/********************************************** 
+ * Class EGS_PlanarFluence Implementation *
+***********************************************/
+
+
+EGS_PlanarFluence::EGS_PlanarFluence(const string &Name, EGS_ObjectFactory *f) :
+    EGS_FluenceScoring(Name,f), hits_field(false), Nx(1), Ny(1)
+{
+    otype = "EGS_PlanarFluence";
+    m_midpoint = EGS_Vector(0,0,5);
+    m_R  =  5;
+    m_R2 = 25;
+    field_type = circle; Area = M_PI * m_R2;
+    m_normal = EGS_Vector(0,0,1);
+    m_d = m_normal*m_midpoint;
+}
+
+/*! Destructor.  */
+EGS_PlanarFluence::~EGS_PlanarFluence(){
+
+   if( flu ) {
+       for(int j=0; j<Nx*Ny; j++) {delete flu[j];}
+   }
+   if( flu_p ) {
+       for(int j=0; j<Nx*Ny; j++) {delete flu_p[j];}
+   }
+}
+
+void EGS_PlanarFluence::setApplication(EGS_Application *App) {
+
+    EGS_AusgabObject::setApplication(App);
+
+    if( !app ) return;
+
+    /*********************************************************** 
+       Defaults to charge of application source. This applies most 
+       of the time, except for bremsstrahlung targets and multiple 
+       particle sources such as radiactive sources. 
+       Can be changed via 'source particle' input.
+    *************************************************************/
+    if ( source_charge == unknown )
+         source_charge = (ParticleType)app->sourceCharge();
+
+
+   // Get the number of regions in the geometry.
+    nreg = app->getnRegions();
+
+    /* Initialize arrays with defaults */
+    for (int j=0; j<nreg; j++) {
+
+        if ( !score_in_all_regions )
+           is_sensitive.push_back(false);
+        else
+           is_sensitive.push_back(true);
+
+        is_source.push_back(false);
+    }
+
+    /* Flag scoring and source regions*/
+    setUpRegionFlags();
+
+    /* Setup fluence scoring arrays, Nx*Ny = 1 for the circle */
+    fluT = new EGS_ScoringArray(Nx*Ny);
+    if ( score_spe ){
+       flu  = new EGS_ScoringArray* [Nx*Ny];
+       for (int j = 0; j < Nx*Ny; j++) {
+           flu[j] = new EGS_ScoringArray(flu_nbin);
+       }
+    }
+
+    if ( score_primaries ){
+       fluT_p = new EGS_ScoringArray(Nx*Ny);
+       if ( score_spe ){
+          flu_p  = new EGS_ScoringArray* [Nx*Ny];
+          for (int j = 0; j < Nx*Ny; j++){ 
+              flu_p[j] = new EGS_ScoringArray(flu_nbin);
+          }           
+       }
+    }
+
+    describeMe();
+}
+
+void EGS_PlanarFluence::initScoring(EGS_Input *inp) {
+
+    EGS_Input *pScoringInput;
+
+    if( !inp ) {
+        egsFatal("AO type %s: null input?\n",otype.c_str()); return;
+    }
+    else{
+       pScoringInput = inp->takeInputItem("planar scoring");
```

Comment:
This input is not described in the documentation.

Reply:
Working on it ....

Reply:
@rtownson is this "resolved"?